syntax = "proto3";

option java_multiple_files = true;
option java_outer_classname = "MatriculasProto";

// interface de servi√ßo
service ControleNotas {
  rpc AdicionarMatricula (MatriculaRequest) returns (MatriculaResponse) {}
  rpc AlterarNota (MatriculaRequest) returns (MatriculaResponse) {}
  rpc AlterarFaltas (MatriculaRequest) returns (MatriculaResponse) {}
  rpc ListarAlunos (ListarAlunosRequest) returns (ListarAlunosResponse) {}
  rpc ListarDisciplinasAluno (BoletimRequest) returns (BoletimResponse) {}
  rpc ListarDisciplinasCurso (ListarDisciplinasCursoRequest) returns (ListarDisciplinasCursoResponse) {}
}

message MatriculaRequest {
    int32 ra = 1;
	string cod_disciplina = 2;
	int32 ano = 3;
	int32 semestre = 4;
	float nota = 5;
	int32 faltas = 6;
}

message MatriculaResponse {
    int32 ra = 1;
	string cod_disciplina = 2;
	int32 ano = 3;
	int32 semestre = 4;
	float nota = 5;
	int32 faltas = 6;
    string mensagem = 7;
}

message ListarAlunosRequest {
    string cod_disciplina = 1;
    int32 ano = 2;
    int32 semestre = 3;
}

message ListarAlunosResponse {
    message Aluno{
        int32 ra = 1;
        string nome = 2;
        int32 periodo = 4;
    }
    repeated Aluno alunos = 1;
    string mensagem = 2;
}

message BoletimRequest {
    int32 ra = 1;
    int32 ano = 2;
    int32 semestre = 3;
}

message BoletimResponse {
    message DisciplinaAlunos{
        int32 ra = 1;
        string cod_disciplina = 2;
        float nota = 3;
        int32 faltas = 4;  
    }
    repeated DisciplinaAlunos disciplinas = 1;
    string mensagem = 2;
}

message ListarDisciplinasCursoRequest {
    int32 cod_curso = 1;
}

message ListarDisciplinasCursoResponse {
    message DisciplinaCurso{
        string cod_disciplina = 1;
        string nome = 2;
        string professor = 3;
    }
    repeated DisciplinaCurso disciplinas = 1;
    string nome = 2;
    string mensagem = 3;
}